apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Types/UserRoleAuditedChangeAuditLogDtoType.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.WebApi.Clients.MarketParticipant.v1;
+using HotChocolate.Types;
+
+namespace Energinet.DataHub.WebApi.GraphQL;
+
+public sealed class UserRoleAuditedChangeAuditLogDtoType : ObjectType<UserRoleAuditedChangeAuditLogDto>
+{
+    protected override void Configure(IObjectTypeDescriptor<UserRoleAuditedChangeAuditLogDto> descriptor)
+    {
+        descriptor
+            .Field(f => f.AuditIdentityId)
+            .Name("auditedBy")
+            .Resolve(async (ctx, ct) =>
+            {
+                var parent = ctx.Parent<UserRoleAuditedChangeAuditLogDto>();
+                var auditIdentity = await ctx
+                    .Service<IMarketParticipantClient_V1>()
+                    .AuditIdentityAsync(parent.AuditIdentityId, ct)
+                    .ConfigureAwait(false);
+
+                return auditIdentity.DisplayName;
+            });
+
+        descriptor
+            .Field("affectedPermissionName")
+            .Resolve(async (ctx, ct) =>
+            {
+                var parent = ctx.Parent<UserRoleAuditedChangeAuditLogDto>();
```

Comment:
Question: I do not how the backend is done. But if it was possible to have a ChangedId property instead of currentValue / previousValue we could avoid this pattern matching